### ISSUE TITLE
我是torch选手，咨询一下GHMLoss的valid mask是怎么产生的

### ISSUE THREAD

**Issue by ShadowTeamCN** [zheng-yuwei/multi-label-classification#5]
改了个torch版本的GHMloss，没太理解valid mask是如何定义的
因此自己写的torch版本吧mask去掉了，但是模型欠拟合了
btw，我是做的多标签文本分类，GHMLoss不知道是不是不适合文本问题

**Comment by zheng-yuwei** [comment]
valid mask用于表示sample的哪些标签是要计算计算loss的，我原来场景是有些sample并不需要计算某些标签的损失。你的场景是所有标签都要计算，所以去掉是可以的。你把acc_sum初始化为1数组，momentum初始化为1，那就退化为一般bce loss了，你可以先这样debug下对不对

**Comment by ShadowTeamCN** [comment]
> valid mask用于表示sample的哪些标签是要计算计算loss的，我原来场景是有些sample并不需要计算某些标签的损失。你的场景是所有标签都要计算，所以去掉是可以的。你把acc_sum初始化为1数组，momentum初始化为1，那就退化为一般bce loss了，你可以先这样debug下对不对

好的，谢谢，我操作操作

**Comment by ShadowTeamCN** [comment]
> valid mask用于表示sample的哪些标签是要计算计算loss的，我原来场景是有些sample并不需要计算某些标签的损失。你的场景是所有标签都要计算，所以去掉是可以的。你把acc_sum初始化为1数组，momentum初始化为1，那就退化为一般bce loss了，你可以先这样debug下对不对

对于这一点还有一点疑问，我之前自己实现bce的是类似 -y_truth * torch.log(y_pred) -(1-y_truth) * torch.log(1-y_pred) 这种，计算的结果和torch原生的结果是比较一致的。 而且在把acc_sum置为1数组，momentum置为1之后，按照我自己实现的bce才正常。

**Comment by zheng-yuwei** [comment]
不太理解你的意思。
你自己实现的bce = pytorch F.binary_cross_entropy_with_logits = 正常bce；
ghm （acc_sum=[1...] + momentum=1 ） + 你自己实现的bce => 正常bce；
ghm （acc_sum=[1...] + momentum=1 ） + pytorch F.binary_cross_entropy_with_logits => 异常bce；
这个意思？？

还有一开始说错了，我复现用的softmax loss，不是bce loss，但含义一样的。代码里其实主要就是：
``` python
cross_entropy = -y_truth * keras.backend.log(y_pred)
loss = cross_entropy / grad_density
```

**Comment by ShadowTeamCN** [comment]
> 不太理解你的意思。
> 你自己实现的bce = pytorch F.binary_cross_entropy_with_logits = 正常bce；
> ghm （acc_sum=[1...] + momentum=1 ） + 你自己实现的bce => 正常bce；
> ghm （acc_sum=[1...] + momentum=1 ） + pytorch F.binary_cross_entropy_with_logits => 异常bce；
> 这个意思？？
> 
> 还有一开始说错了，我复现用的softmax loss，不是bce loss，但含义一样的。代码里其实主要就是：
> 
> ```python
> cross_entropy = -y_truth * keras.backend.log(y_pred)
> loss = cross_entropy / grad_density
> ```

原来如此，之前不知道还有softmax loss 这个东西，我按照原来的softmax loss确实欠拟合，但是自己改成bce就可以了。前文的意思是我自己实现的bce和官方的bce结果是一样的。
最终效果没什么提升，好处是取得best f1的阈值在0.5左右，之前在0.2～0.3
最后感谢作者答疑解惑～
